tests/state_migration_test.go

```diff
@@ -55,15 +57,91 @@ func TestMigration_ContinuesRestartAndMigration(t *testing.T) {
 		newPathKey := append([]byte("databaseDirectory"), common.Int64ToByteBigEndian(uint64(database.StateTrieDB))...)
 		newStateTriePath, err := node.ChainDB().GetMiscDB().Get(newPathKey)
 		assert.NoError(t, err)
-		assert.NotEqual(t, stateTriePath, newStateTriePath)
+		assert.NotEqual(t, stateTriePath, newStateTriePath, "migration failed")
 		stateTriePath = newStateTriePath
 	}
 
 	stopNode(t, fullNode)
 }
 
-// if migration status is set on miscDB and a node is restarted, migration should start
+// state trie DB should be determined by the values of miscDB
 func TestMigration_StartMigrationByMiscDB(t *testing.T) {
+	fullNode, node, validator, _, workspace, _, _, _ := newSimpleBlockchain(t, 10)
+	defer os.RemoveAll(workspace)
+
+	stateTriePathKey := append([]byte("databaseDirectory"), common.Int64ToByteBigEndian(uint64(database.StateTrieDB))...)
+
+	// use the default StateTrie DB if it is not set on miscDB
+	{
+		key, err := node.ChainDB().GetMiscDB().Get(stateTriePathKey)
+		assert.Error(t, err)
+		assert.Len(t, key, 0)
+		fullNode, node = checkStatTrieDB(t, fullNode, node, validator, workspace, "statetrie")
+	}
+
+	// after migration, use the created DB
+	{
+		startMigration(t, node)
+		time.Sleep(time.Second)
+		newDBPath2, err := node.ChainDB().GetMiscDB().Get(stateTriePathKey)
```

Comment:
Don't we need to check `newDBPath2` is different with `key`?

Reply:
@KimKyungup The checking is not ment for this test. After I thought, the test is not for state migration directly. So I erased the migration part.